metrics/engine/ingester_test.go

```diff
@@ -96,6 +98,36 @@ func TestIngesterOutputFlushSubmetrics(t *testing.T) {
 	assert.IsType(t, &metrics.GaugeSink{}, metric.Sink)
 }
 
+func TestIngesterOutputWarnTimeSeriesLimit(t *testing.T) {
+	t.Parallel()
+
+	piState := newTestPreInitState(t)
+	testMetric, err := piState.Registry.NewMetric("test_metric", metrics.Gauge)
+	require.NoError(t, err)
+
+	logger, hook := testutils.NewMemLogger()
+
+	ingester := outputIngester{
+		logger:          logger,
+		seenTimeSeries:  make(map[metrics.TimeSeries]struct{}),
+		timeSeriesLimit: 1,
+	}
+	ingester.seenTimeSeries[metrics.TimeSeries{
+		Metric: testMetric,
+		Tags:   piState.Registry.RootTagSet().With("a", "2"),
+	}] = struct{}{}
+	ingester.warnTimeSeriesLimit()
```

Comment:
I think this test is OK as a `warnTimeSeriesLimit()` unit test, but you're not testing that `flushMetrics()` will actually run this check and emit the warning.

So I'd rather see a higher-level test at the output level, but feel free to keep this one as well.

Reply:
[This](https://github.com/grafana/k6/blob/5c7058f0ddde0b3fab8dacca62fc68aa98acbd27/metrics/engine/ingester.go#L108-L117) is still not covered.

Can you add another test that calls `flushMetrics()` and confirms the warning is logged?

Reply:
Added